src/codegen/events/solana.rs

```diff
@@ -46,18 +46,15 @@ impl EventEmitter for SolanaEventEmitter<'_> {
         let mut to_be_encoded: Vec<Expression> = vec![discriminator];
         to_be_encoded.append(&mut codegen_args);
 
-        let mut encoder = create_encoder(self.ns, false);
         let (abi_encoded, abi_encoded_size) =
-            encoder.abi_encode(&self.loc, to_be_encoded, self.ns, vartab, cfg);
+            abi_encode(&self.loc, to_be_encoded, self.ns, vartab, cfg, false);
 
         cfg.add(
             vartab,
             Instr::EmitEvent {
                 event_no: self.event_no,
                 data: vec![abi_encoded, abi_encoded_size],
```

Comment:
Now that you've removed the unused fields from this instructions, there should be different fields for data and size.

Reply:
In addition, `data` does not need to be a vector anymore.

Reply:
Good point!